### PR TITLE
Nonce Error Handling

### PR DESCRIPTION
**Pull request by praetoriansentry**
# Description

There are some scenarios in testing where the transaction fails in such a way that the nonce shouldn't be advanced. This code checks for one such scenario and keeps the nonce the same if that situation occurs.

This PR is also adding some basic summarization at the end of a run even if we don't want to do the full `--summary` process which can take a long time to produce.

image

### CODE REVIEW

**Review by leovct — APPROVED** [review]
lgtm

File: cmd/loadtest/loadtest.go [hunk 1]
```diff
@@ -758,15 +767,48 @@ func mainLoop(ctx context.Context, c *ethclient.Client, rpc *ethrpc.Client) erro
...
-		err = summarizeTransactions(ctx, c, rpc, startBlockNumber, startNonce, currentNonce)
+		err = summarizeTransactions(ctx, c, rpc, startBlockNumber, startNonce, finalBlockNumber, currentNonce)
 		if err != nil {
 			log.Error().Err(err).Msg("There was an issue creating the load test summary")
 		}
 	}
 	return nil
 }
 
+func lightSummary(ctx context.Context, c *ethclient.Client, rpc *ethrpc.Client, startBlockNumber, startNonce, endBlockNumber, endNonce uint64) {
```
Comment (rebelArtists) [inline cmd/loadtest/loadtest.go:786]: nice

**Review by rebelArtists — APPROVED** [review]
lgtm

File: cmd/loadtest/loadtest.go [hunk 2]
```diff
@@ -588,7 +589,7 @@ func mainLoop(ctx context.Context, c *ethclient.Client, rpc *ethrpc.Client) erro
 		tops.Nonce = new(big.Int).SetUint64(currentNonce)
 		tops.GasLimit = 10000000
 		tops = configureTransactOpts(tops)
-		_, err = erc20Contract.Mint(tops, new(big.Int).SetUint64(1_000_000_000_000))
+		_, err = erc20Contract.Mint(tops, metrics.UnitMegaether)
```
[the remark this answers is not in the document]
Comment (praetoriansentry) [reply cmd/loadtest/loadtest.go:592]: I believe it pulled it from here: https://github.com/ethereumbook/ethereumbook/blob/05f0dfe6c41635ac85527a60c06ac5389d8006e7/02intro.asciidoc#ether_denominations

[the remark this answers is not in the document]
Comment (rebelArtists) [reply cmd/loadtest/loadtest.go:592]: ah got it

**Pull request merged**